theseus/optimizer/nonlinear/levenberg_marquardt.py

```diff
@@ -2,23 +2,42 @@
 #
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-
-from typing import Any, Dict, Optional, Type
+import warnings
+from typing import Any, Dict, List, Optional, Sequence, Type, Union
 
 import torch
 
 from theseus.core import Objective
-from theseus.optimizer import Linearization
-from theseus.optimizer.linear import DenseSolver, LinearSolver, LUCudaSparseSolver
+from theseus.optimizer import DenseLinearization, Linearization
+from theseus.optimizer.linear import (
+    CholeskyDenseSolver,
+    LinearSolver,
+    LUCudaSparseSolver,
+    LUDenseSolver,
+)
 
 from .nonlinear_least_squares import NonlinearLeastSquares
 
-_LM_ALLOWED_ELLIPS_DAMP_SOLVERS = [DenseSolver, LUCudaSparseSolver]
+_LM_ALLOWED_ELLIPS_DAMP_SOLVERS: List[Type[LinearSolver]] = [
+    CholeskyDenseSolver,
+    LUDenseSolver,
```

Comment:
Since all `DenseSolver` in general are supported is it necessary to qualify the two current dense solvers explicitly?

Reply:
This was to avoid some `mypy` annoyance, because apparently `Type` is meant to be used only for non-abstract  classes. I found multiple `mypy` issues complaining about this limitation, and some workarounds that didn't seem to quite apply to our case.